### ISSUE TITLE
`.vue` files under `node_modules` are never virtualised, so tsgo parses raw SFC text

### ISSUE THREAD

**Issue by maiolica** [johnsoncodehk/typescript-native-bridge#47]
Repro: https://github.com/maiolica/tnb-node-modules-sfc-repro

## Summary

When a dependency ships raw `.vue` source rather than compiled output, importing those SFCs fails
under TNB with `TS2306: ... is not a module` (or `TS1192: ... has no default export`, depending on
how the SFC is written). Stock `typescript@6.0.3` with `vue-tsc@3.3.9` checks the same imports fine.

Worse than the spurious errors, real diagnostics against those components go missing. A wrong prop
type is reported by stock and not by TNB, hidden behind the resolution failure.

Not the same as #31 (SFC inside the project, missing from `include`, which is fixed and passes for
me on `bridge.7`) or #26 (named type exports from a package entry point). I didn't find anything
covering SFCs inside `node_modules`.

## Repro

Three byte-identical SFCs, imported from three locations, so location is the only variable.
`src/Render.vue` passes `:label="123"` to each, where `label` is typed `string`.

| Diagnostic | stock | TNB `6.0.3-bridge.7.tsgo.7.0.2` |
| --- | --- | --- |
| `TS2306` on the `node_modules` SFC | none | 2, one per importing file |
| wrong prop on the `node_modules` SFC | 1 | 0, lost |
| wrong prop on an SFC outside `node_modules` | 1 | 1 |
| wrong prop on an SFC in `src` | 1 | 1 |

Also on `bridge.7`:

| Variant | Result |
| --- | --- |
| SFC outside `node_modules` and outside `include` | fine, so #31's fix is present |
| import a `.vue` that doesn't exist on disk | fine, the ambient `declare module '*.vue'` applies |
| `allowArbitraryExtensions: false` | no change |
| vendor SFC added to `include` explicitly | no change |

Same thing at scale on a large private Vue codebase whose component library is consumed as source:
thousands of these errors under TNB, none under stock, all gone with the change below. A few
unrelated pre-existing divergences remain there, which I'm not raising here.

## Cause

`lib/_tsc.js`, line 106537:

```js
function isOverlayCandidatePath(fileName) {
  if (isBundledLibPath(fileName)) return false;
  return !fileName.includes("/lib.") && !fileName.includes("/node_modules/");
}
```

The overlay-collection loops skip these paths (`if (!isOverlayCandidatePath(fn)) continue;`) and the
other call sites bail out equivalently, so the host's generated TypeScript for an SFC under
`node_modules` is never pushed across. tsgo then reads the file off disk and parses `<script setup>`
as TypeScript, hence "is not a module". Outside `node_modules` the content flows normally, which
matches the isolation table.

The repro includes a probe that confirms this directly. `vendor-ui/RawModule.vue` is not a valid SFC:
its whole content is `export default { name: "raw" };`. Under TNB that import is clean while the real
SFC beside it fails, so the raw disk text is what reaches the checker. It also explains the
`TS2306` / `TS1192` split. Raw text with no imports or exports parses as a script, so "not a module".
Raw `<script setup>` text starting with imports parses as a module with no default export.

Letting extra-extension files through restores exact parity with stock on the repro:

```js
function isOverlayCandidatePath(fileName) {
  if (isBundledLibPath(fileName)) return false;
  if (isExtraExtensionFileName(fileName)) return !fileName.includes("/lib.");
  return !fileName.includes("/lib.") && !fileName.includes("/node_modules/");
}
```

That keeps the performance intent, since `.js` and `.d.ts` under `node_modules` still skip the
overlay.

Caveats:

- I patched the shipped `lib/_tsc.js` bundle rather than building from source, so treat that diff as
  a demonstration, not a tested patch. `lib/typescript.js` has the same function and I didn't
  exercise it, since the `vue-tsc` CLI doesn't load it.
- I didn't test tsserver. If it shares the predicate, hover and go-to-definition on these components
  are presumably affected too.
- Presumably the same for any host-virtualised extension shipped inside `node_modules`, though I only
  tried `.vue`.

## Expected

A dependency can ship source in a host-virtualised extension, so host-provided content should reach
the checker whether or not the path contains `node_modules`, and real errors against those components
shouldn't be dropped.

## Environment

TNB `6.0.3-bridge.7.tsgo.7.0.2` via a pnpm override on `typescript`, `vue-tsc` 3.3.9, `vue` 3.5.24,
baseline `typescript` 6.0.3, macOS arm64, Node 24.18.1.

Happy to run instrumented builds if there's a probe that would help.
